### PR TITLE
parser: finish the restore function of all ast node

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
Most of the restore function has been completed. but some test cases didn't enable.

this pr switch on all of the test cases in `parser_test.go` and fix some bug exposed in testing.

**after pingcap/tidb#9302 merged, i will update the version of tidb dependency, then all the UT will pass**

Issue: pingcap/tidb#8532

### Check List 

Tests 

 - Integration test
switch on many tests in `parser_test.go`

### CODE REVIEW

**Comment by leoppro** [comment]
@GregoryIan @kennytm PTAL

File: ast/ddl.go [hunk 1]
```diff
@@ -1280,13 +1281,15 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		}
 	case TableOptionStatsPersistent:
 		// TODO: not support
+		ctx.WriteKeyWord("STATS_PERSISTENT = DEFAULT")
```
Comment (kennytm) [inline ast/ddl.go]:
```suggestion
		ctx.WriteKeyWord("STATS_PERSISTENT")
		ctx.WritePlain(" = ")
		ctx.WriteKeyWord("DEFAULT")
```

File: ast/ddl.go [hunk 2]
```diff
@@ -1280,13 +1281,15 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
...
 		ctx.WritePlain(" /* TableOptionStatsPersistent is not supported */ ")
 	case TableOptionShardRowID:
 		ctx.WriteKeyWord("SHARD_ROW_ID_BITS ")
 		ctx.WritePlain("= ")
 		ctx.WritePlainf("%d", n.UintValue)
 	case TableOptionPackKeys:
 		// TODO: not support
+		ctx.WriteKeyWord("PACK_KEYS = DEFAULT")
```
Comment (kennytm) [inline ast/ddl.go]: (ditto)

File: ast/ddl.go [hunk 3]
```diff
@@ -1280,13 +1281,15 @@ func (n *TableOption) Restore(ctx *RestoreCtx) error {
 		}
 	case TableOptionStatsPersistent:
 		// TODO: not support
+		ctx.WriteKeyWord("STATS_PERSISTENT = DEFAULT")
 		ctx.WritePlain(" /* TableOptionStatsPersistent is not supported */ ")
```
Comment (kennytm) [inline ast/ddl.go]: Do we still need this?

File: ast/misc.go [hunk 4]
```diff
@@ -1232,9 +1246,7 @@ type PrivElem struct {
 func (n *PrivElem) Restore(ctx *RestoreCtx) error {
 	switch n.Priv {
 	case 0:
-		// Do nothing for types that have no effect.
-		// Actually this should not happen since there is no way to determine its type.
-		return errors.New("Cannot determine privilege type")
+		ctx.WritePlainf("/* UNSUPPORTED TYPE */")
```
Comment (kennytm) [inline ast/misc.go]:
```suggestion
		ctx.WritePlain("/* UNSUPPORTED TYPE */")
```

File: ast/ddl.go (hunk 3 above)
Comment (leoppro) [reply ast/ddl.go]: yes, it's necessary.
The following two sql corresponds to two different ast tree:
```
create table t (c int) STATS_PERSISTENT = default

create table t (c int);
```
To ensure that the two ast trees are equal, we have to add `STATS_PERSISTENT = default`

Comment (kennytm) [reply ast/ddl.go]: I mean is the comment `/* TableOptionStatsPersistent is not supported */` still needed to be printed out?

Comment (leoppro) [reply ast/ddl.go]: yes, It's used to remind people that `STATS_PERSISTENT = default` is not the right result.

**Comment by GregoryIan** [comment]
LGTM

**Comment by leoppro** [comment]
@kennytm  PTAL

**Pull request merged**